File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        d = {')':'(', ']':'[', '}':'{'}
        stack = []

        for char in s:
            if char not in d:
                stack.append(char)
            else:
                if not stack:
                    return False
                else:
                    if stack.pop() != d[char]:
                        return False
                    
        
        return not stack
                    
                

                
                
                    

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
#include<iostream>
#include<stack>
#include<unordered_map>
#include<vector>
#include<string>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        unordered_map <char, char> dict;
        dict = {{')','('}, {']','['}, {'}', '{'}};
        stack<char> stack;

        for(int i=0; i < s.length(); i++){
            if(dict.find(s[i])==dict.end()){
                stack.push(s[i]);}
            else{
                if(stack.empty() || stack.top() != dict[s[i]]){
                    return false;
                }
                stack.pop();
            }
        
            
            


        }
        
        return stack.empty();
        
    }
};